### PR TITLE
Automatically add Content-Length parameter if needed

### LINKED ISSUE #160: Add Content-Length parameter when Stream request body is used
Right now we're having Swagger authors add a Content-Length parameter to their specs. This isn't quite right. Since it doesn't look like we're going to support chunked upload scenarios, we might want to just have the generator automatically add a Content-Length parameter when a `Flowable<ByteBuffer>` is going to be used for the request body.

### PR DESCRIPTION
**Pull request by RikkiGibson**
Fixes #160.

### CODE REVIEW

File: src/JavaCodeGenerator.cs [hunk 1]
```diff
@@ -1066,6 +1066,30 @@ private static RestAPIMethod ParseRestAPIMethod(AutoRestMethod autoRestMethod, J
...
+                contentLengthParameter.Name = "contentLength";
+                contentLengthParameter.Documentation = "The content length";
+                contentLengthParameter.ModelType = DependencyInjection.New<AutoRestPrimaryType>(AutoRestKnownPrimaryType.Long);
+
+                // Add the Content-Length parameter before the body parameter
+                autoRestMethodParameters.Insert(streamBodyParameterIndex, contentLengthParameter);
+            }
+
+            autoRestMethod.ClearParameters();
+            autoRestMethod.AddRange(autoRestMethodParameters);
```
Comment (daschult) [inline src/JavaCodeGenerator.cs]: You only need to do these two lines if you insert the contentLengthParameter, right? In that case, this should probably be moved inside the if-statement. If you didn't insert the contentLengthParameter, then there's no need to clear the autoRestMethod's parameters.

File: test/vanilla/src/main/java/fixtures/bodyformdata/Formdatas.java [hunk 2]
```diff
@@ -71,39 +71,43 @@
...
      *
+     * @param contentLength The content length.
      * @param fileContent File to upload.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      * @return the Flowable&lt;ByteBuffer&gt; object if successful.
      */
-    Flowable<ByteBuffer> uploadFileViaBody(@NonNull Flowable<ByteBuffer> fileContent);
+    Flowable<ByteBuffer> uploadFileViaBody(@NonNull long contentLength, @NonNull Flowable<ByteBuffer> fileContent);
```
Comment (daschult) [inline test/vanilla/src/main/java/fixtures/bodyformdata/Formdatas.java]:
```java
@NonNull long contentLength
```
I guess this is a bug in the generator. I think it's pretty clear that long will never be null.

Comment (RikkiGibson) [reply test/vanilla/src/main/java/fixtures/bodyformdata/Formdatas.java]: I guess it's not needed, but I also don't foresee it causing users any pain. Are primitives in general having this annotation applied?

**Comment by RikkiGibson** [comment]
Moving to #178 so we can have CI

File: src/JavaCodeGenerator.cs (hunk 1 above)
Comment (RikkiGibson) [reply src/JavaCodeGenerator.cs]: Sure, seems better. Will fix in the other branch.

File: test/vanilla/src/main/java/fixtures/bodyformdata/Formdatas.java (hunk 2 above)
Comment (daschult) [reply test/vanilla/src/main/java/fixtures/bodyformdata/Formdatas.java]: It looks like they are. I'm surprised, because I thought I had handled this case.

**Pull request closed without merging**